packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -94,19 +97,43 @@ class RenderAndroidView extends RenderBox {
 
   /// Which gestures should be forwarded to the Android view.
   ///
-  /// The gesture recognizers on this list participate in the gesture arena for each pointer
-  /// that was put down on the render box. If any of the recognizers on this list wins the
+  /// Gesture recognizers created by factories in this set participate in the gesture arena for each
+  /// pointer that was put down on the render box. If any of the recognizers on this list wins the
   /// gesture arena, the entire pointer event sequence starting from the pointer down event
   /// will be dispatched to the Android view.
-  set gestureRecognizers(List<OneSequenceGestureRecognizer> recognizers) {
-    assert(recognizers != null);
-    if (recognizers == _gestureRecognizer?.gestureRecognizers) {
+  ///
+  /// `gestureRecognizers` must not contain more than one factory with the same [Factory.type].
+  ///
+  /// Setting a new set of gesture recognizer factories with the same [Factory.type]s as the current
+  /// set is ignored.
+  ///
+  /// Setting a new set of gesture recognizer factories with different [Factory.type]s
+  /// (e.g `gestureRecognizers.map((f) => f.type).toSet()` is different, will reject any active
+  /// gesture arena.
+  void setGestureRecognizers(Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers) {
+    assert(gestureRecognizers != null);
+    assert(_factoriesTypeSet(gestureRecognizers).length == gestureRecognizers.length);
+    if (_factoryTypesSetEquals(gestureRecognizers, _gestureRecognizer?.gestureRecognizerFactories)) {
       return;
     }
     _gestureRecognizer?.dispose();
-    _gestureRecognizer = _AndroidViewGestureRecognizer(_motionEventsDispatcher, recognizers);
+    _gestureRecognizer = _AndroidViewGestureRecognizer(_motionEventsDispatcher, gestureRecognizers);
   }
 
+  static bool _factoryTypesSetEquals<T>(Set<Factory<T>> a, Set<Factory<T>> b) {
+    if (a == b) {
+      return true;
+    }
+    if (a == null ||  b == null) {
+      return false;
+    }
+    return setEquals(_factoriesTypeSet(a), _factoriesTypeSet(b));
+  }
+
+  static Set<Type> _factoriesTypeSet<T>(Set<Factory<T>> factories) =>
+      factories.map<Type>((Factory<T> factory) => factory.type).toSet();
```

Comment:
i'm amused that you used => here (even though the whole thing doesn't fit on one line), but you used `{ }` for the Factory.toString method, even though that could easily fit on one line.